src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -533,10 +533,10 @@ BookStep<TIn, TOut, TDerived>::forEachOffer (
                 // Issuer requires authorization.  See if offer owner has that.
                 auto const& ownerID = offer.owner();
                 auto const authFlag =
-                    ownerID > issuerID ? lsfHighAuth : lsfLowAuth;
+                    ownerID > issuerID ? lsfLowAuth : lsfHighAuth;
 
                 auto const line = afView.read (keylet::line (
-                    ownerID, issuerID, offer.issueOut().currency));
+                    ownerID, issuerID, offer.issueIn().currency));
```

Comment:
Great catch you two.  Checking for XRP at the top of the condition becomes an optimization because then we don't need to compute the `keyless::account`.  Is this what you had in mind?
```
        // Make sure offer owner has authorization to own IOUs from issuer.
        // An account can always own XRP or their own IOUs.
        if (flowCross &&
            (offer.issueIn().currency != xrpCurrency()) &&
            (offer.owner() != offer.issueIn().account))
        {
            ...
```